chore_: cleaning the accounts manager

saledjenic opened the pull request:
The following functions have been removed from the account package:
- `Recover`
- `CanRecover`
- `Sign`

and the rest of the code updates to use the same functions from the personal api package,
where we already have those functions.

`MockedPersonalAPI` within the personal package was added as the easiest way of fixing tests (basically it does what was done in the `AccountManagerMock`, cause any other approach requires a lot of changes in communities/messenger tests.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (40)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 240837a0 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6598/1/) | 2025-05-22 15:29:30 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250522-152638-240837-pr6598.aar) |
| :heavy_check_mark: | 240837a0 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6598/1/) | 2025-05-22 15:30:00 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250522-152640-240837-pr6598.zip) |
| :heavy_check_mark: | 240837a0 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6598/1/) | 2025-05-22 15:31:54 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250522-152643-240837-pr6598.zip) |
| :heavy_check_mark: | 240837a0 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6598/1/) | 2025-05-22 15:32:45 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250522-152641-240837-pr6598.zip) |
| :heavy_check_mark: | 240837a0 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6598/1/) | 2025-05-22 15:32:54 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250522-152642-240837-pr6598.zip) |
| :heavy_check_mark: | 240837a0 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6598/1/) | 2025-05-22 15:33:04 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250522-152647-240837-pr6598.zip) |
| :heavy_check_mark: | 240837a0 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/1/) | 2025-05-22 15:38:29 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/1//consoleText) |
| :heavy_multiplication_x: | 240837a0 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/1/) | 2025-05-22 16:01:08 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3f645bc3 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6598/2/) | 2025-05-22 15:32:24 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250522-152934-3f645b-pr6598.aar) |
| :heavy_check_mark: | 3f645bc3 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6598/2/) | 2025-05-22 15:33:22 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250522-153005-3f645b-pr6598.zip) |
| :heavy_check_mark: | 3f645bc3 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6598/2/) | 2025-05-22 15:37:02 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250522-153259-3f645b-pr6598.zip) |
| :heavy_check_mark: | 3f645bc3 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6598/2/) | 2025-05-22 15:37:10 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250522-153203-3f645b-pr6598.zip) |
| :heavy_check_mark: | 3f645bc3 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6598/2/) | 2025-05-22 15:37:27 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250522-153307-3f645b-pr6598.zip) |
| :heavy_check_mark: | 3f645bc3 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6598/2/) | 2025-05-22 15:37:54 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250522-153249-3f645b-pr6598.zip) |
| :heavy_check_mark: | 3f645bc3 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/2/) | 2025-05-22 15:48:17 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/2//consoleText) |
| :heavy_multiplication_x: | 3f645bc3 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/2/) | 2025-05-22 16:34:10 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4da77fa7 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6598/3/) | 2025-05-23 08:57:42 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250523-085448-4da77f-pr6598.aar) |
| :heavy_check_mark: | 4da77fa7 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6598/3/) | 2025-05-23 08:58:09 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250523-085449-4da77f-pr6598.zip) |
| :heavy_check_mark: | 4da77fa7 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6598/3/) | 2025-05-23 08:59:47 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250523-085457-4da77f-pr6598.zip) |
| :heavy_check_mark: | 4da77fa7 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6598/3/) | 2025-05-23 09:00:04 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250523-085454-4da77f-pr6598.zip) |
| :heavy_check_mark: | 4da77fa7 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6598/3/) | 2025-05-23 09:00:13 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250523-085452-4da77f-pr6598.zip) |
| :heavy_check_mark: | 4da77fa7 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6598/3/) | 2025-05-23 09:02:22 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250523-085450-4da77f-pr6598.zip) |
| :heavy_multiplication_x: | 4da77fa7 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/3/) | 2025-05-23 09:06:31 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/3//consoleText) |
| :heavy_multiplication_x: | 4da77fa7 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/3/) | 2025-05-23 09:31:07 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 20a9c459 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6598/4/) | 2025-05-23 09:01:22 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250523-085827-20a9c4-pr6598.aar) |
| :heavy_check_mark: | 20a9c459 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6598/4/) | 2025-05-23 09:01:48 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250523-085827-20a9c4-pr6598.zip) |
| :heavy_check_mark: | 20a9c459 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6598/4/) | 2025-05-23 09:04:21 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250523-085950-20a9c4-pr6598.zip) |
| :heavy_check_mark: | 20a9c459 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6598/4/) | 2025-05-23 09:04:59 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250523-090029-20a9c4-pr6598.zip) |
| :heavy_check_mark: | 20a9c459 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6598/4/) | 2025-05-23 09:05:15 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250523-090009-20a9c4-pr6598.zip) |
| :heavy_check_mark: | 20a9c459 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6598/4/) | 2025-05-23 09:07:39 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250523-090227-20a9c4-pr6598.zip) |
| :heavy_check_mark: | 20a9c459 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/4/) | 2025-05-23 09:16:09 | ~9 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/4//consoleText) |
| :heavy_multiplication_x: | 20a9c459 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/4/) | 2025-05-23 10:05:27 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 05bd6ab2 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6598/5/) | 2025-05-23 11:24:03 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250523-112102-05bd6a-pr6598.aar) |
| :heavy_check_mark: | 05bd6ab2 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6598/5/) | 2025-05-23 11:24:27 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250523-112103-05bd6a-pr6598.zip) |
| :heavy_check_mark: | 05bd6ab2 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6598/5/) | 2025-05-23 11:25:57 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250523-112111-05bd6a-pr6598.zip) |
| :heavy_check_mark: | 05bd6ab2 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6598/5/) | 2025-05-23 11:26:15 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250523-112107-05bd6a-pr6598.zip) |
| :heavy_check_mark: | 05bd6ab2 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6598/5/) | 2025-05-23 11:26:25 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250523-112107-05bd6a-pr6598.zip) |
| :heavy_check_mark: | 05bd6ab2 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6598/5/) | 2025-05-23 11:28:42 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250523-112104-05bd6a-pr6598.zip) |
| :heavy_check_mark: | 05bd6ab2 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/5/) | 2025-05-23 11:32:20 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/5//consoleText) |
| :heavy_check_mark: | 05bd6ab2 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/5/) | 2025-05-23 11:57:47 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/5//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 6662a9be | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6598/6/) | 2025-05-23 12:18:56 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250523-121542-6662a9-pr6598.zip) |
| :heavy_check_mark: | 6662a9be | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6598/6/) | 2025-05-23 12:19:04 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250523-121542-6662a9-pr6598.aar) |
| :heavy_check_mark: | 6662a9be | [#6](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6598/6/) | 2025-05-23 12:20:18 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250523-121551-6662a9-pr6598.zip) |
| :heavy_check_mark: | 6662a9be | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6598/6/) | 2025-05-23 12:20:48 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250523-121546-6662a9-pr6598.zip) |
| :heavy_check_mark: | 6662a9be | [#6](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6598/6/) | 2025-05-23 12:21:23 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250523-121547-6662a9-pr6598.zip) |
| :heavy_check_mark: | 6662a9be | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6598/6/) | 2025-05-23 12:22:27 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250523-121543-6662a9-pr6598.zip) |
| :heavy_check_mark: | 6662a9be | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/6/) | 2025-05-23 12:25:50 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/6//consoleText) |
| :heavy_check_mark: | 6662a9be | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/6/) | 2025-05-23 12:51:39 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a58742f1 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6598/7/) | 2025-05-28 06:01:37 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250528-055843-a58742-pr6598.aar) |
| :heavy_check_mark: | a58742f1 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6598/7/) | 2025-05-28 06:01:59 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250528-055844-a58742-pr6598.zip) |
| :heavy_check_mark: | a58742f1 | [#7](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6598/7/) | 2025-05-28 06:03:31 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250528-055852-a58742-pr6598.zip) |
| :heavy_check_mark: | a58742f1 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6598/7/) | 2025-05-28 06:03:57 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-055848-a58742-pr6598.zip) |
| :heavy_check_mark: | a58742f1 | [#7](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6598/7/) | 2025-05-28 06:03:58 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250528-055848-a58742-pr6598.zip) |
| :heavy_check_mark: | a58742f1 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6598/7/) | 2025-05-28 06:04:39 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250528-055845-a58742-pr6598.zip) |
| :heavy_check_mark: | a58742f1 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/7/) | 2025-05-28 06:09:29 | ~10 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6598/7//consoleText) |
| :heavy_check_mark: | a58742f1 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/7/) | 2025-05-28 06:34:29 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6598/7//consoleText) |

saledjenic committed:
chore_: cleaning the accounts manager

The following functions have been removed from the account package:
- `Recover`
- `CanRecover`
- `Sign`

and the rest of the code update to use the same functions from the personal api package,
where we already have those functions.

`MockedPersonalAPI` within the personal package was added as the easiest way of fixing
tests (basically it does what was done in the `AccountManagerMock`, cause any other approach
requires a lot of changes in communities/messenger tests.

igor-sirotin commented on services/personal/api.go:
```diff
@@ -1,91 +1,91 @@
-type RecoverParams struct {
-	Message   string `json:"message"`
-	Signature string `json:"signature"`
+// This interface is used to mock the personal API for testing purposes
+// without it in many places in tests we would need to create verifiedAccount (including a proper account manager setting) and
+// that will also require pdating all fake addresses/signatures etc. we used in communities/messenger tests.
+type PersonalAPIInterface interface {
+	Recover(rpcParams RecoverParams) (addr types.Address, err error)
+	CanRecover(rpcParams RecoverParams, revealedAddress types.Address) (bool, error)
+	Sign(rpcParams SignParams, verifiedAccount *account.SelectedExtKey) (result types.HexBytes, err error)
```
I think this interface should be defined in `Messenger` instead. That would be Go-idiomatic way.
Because that's the requirement of `Messenger`: provide something that implements such interface.

saledjenic replied:
Correct. Moving this there.

saledjenic replied:
@igor-sirotin changes applied.

igor-sirotin commented on services/personal/mocked_api.go:
But this is more or less a duplication of the API itself 🤔 To me it would make more sense to use the real implementation in tests as well. 

I see that it's problematic because you'd need to instantiate the whole `personal/PublicAPI` in tests, which doesn't make sense. But these 3 functions don't have any accounts dependencies, maybe we could extract them as some free utils functions? 

(I know this is a standard `personal` eth api, so please correct me if I'm proposing something wrong)

saledjenic replied:
Well, it was not like that before, cause personal API was using rpcClient, but since it was not needed and the current changes make sign/recover locally, I agree we can move them to some utils/common and will add just a reference to them from here, just in case the client is using endpoints exposed via this API.

@igor-sirotin does it make sense to move them to the wallet service (I guess they should be part of wallet core that we will build the wallet SDK on top of)?

saledjenic replied:
But I cannot actually, cause in that case we will work with a real sing/recover functions and that will require a change of many other community/messenger related files.

igor-sirotin replied:
@saledjenic hmmm, you mean this will break tests? or prod code?

saledjenic replied:
@igor-sirotin yes, talking about tests only. You can try and see. I am not that familiar with messenger/community tests but what I see is that many tests that we have are using some dummy addresses like "0x1", "0x2", "0xsome-random-numbers" and that's the reason why we had this (otherwise we need to provide appropriate verified account, means private key, for each of those addresses):

https://github.com/status-im/status-go/blob/develop/protocol/communities_messenger_helpers_test.go#L47-L57

Since my intention in this PR was not to fix that, but just to remove functions that logically don't belong to account manager from it and since I don't see that change as something that I can easily do, let's do that change in a new PR (ideally by someone who is more familiar with messenger/communities :) ).

igor-sirotin replied:
Opened a PR to try a proper fix: https://github.com/status-im/status-go/pull/6604

igor-sirotin commented:
@saledjenic I like the cleanup btw, thank you 👍 
Just a few minor questions left

igor-sirotin commented on api/geth_backend.go:
```diff
@@ -142,7 +143,7 @@ func (b *GethStatusBackend) initialize() {
 	b.accountManager = accountManager
 	b.transactor = transactor
-	b.personalAPI = personalAPI
+	b.signer = personalAPI
```
```suggestion
	b.signer = signer
```

igor-sirotin reviewed (approved):
Thank you 👍 
I've re-implemented the mock for tests here: https://github.com/status-im/status-go/pull/6604
I guess you can merge your PR, and we can merge my on top later. If it works 😄
